plugin/src/pages/AIPage/index.tsx

```
@@ -0,0 +1,260 @@
+
+  const finalize = useCallback(() => {
+    setIsRunning(false);
+    notify.success('AI rename complete');
+  }, [setIsRunning]);
+
+  // ── Register plugin event listeners ───────────────────────────────────────
+  useEffect(() => {
+    const unsubBatchReady = on<AIBatchReadyHandler>('AI_BATCH_READY', async (groups: AIRenameGroup[]) => {
+      const total = groups.reduce((acc, g) => acc + g.targetNodes.length, 0);
+      setProgress({ done: 0, total });
+      groups.forEach((g) => g.targetNodes.forEach((t) => setRenameStatus(t.nodeId, 'pending')));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)